### PR TITLE
qtgui: Remove more manual memory management

### PR DESCRIPTION
**Pull request by ThomasHabets**
I want to do some more testing today.

I'm not sure why `EyeDisplayPlot` was only freeing the `d_curve_index` `d_ydata`. Was that a mistake, or am I missing something?

### CODE REVIEW

**Review by mormj — APPROVED** [review]
Thanks for the cleanup as always!

**Comment by ThomasHabets** [comment]
I tested a bit more, and seems to work fine.

File: gr-qtgui/lib/timeRasterGlobalData.cc [hunk 1]
```diff
@@ -168,9 +158,9 @@ double TimeRasterData::value(double x, double y) const
 
     double _y = floor(top - y);
     double _loc = _y * (d_cols) + x + d_resid;
-    int location = static_cast<int>(_loc);
+    auto location = static_cast<ssize_t>(_loc);
```
Comment (mbr0wn) [inline gr-qtgui/lib/timeRasterGlobalData.cc:161]: Could have been
```suggestion
    const auto location = static_cast<ssize_t>(_loc);
```

**Review by mbr0wn — APPROVED** [review]
It's so... clean.

File: gr-qtgui/lib/TimeDomainDisplayPlot.cc [hunk 2]
```diff
@@ -90,20 +90,13 @@ class TimeDomainDisplayZoomer : public QwtPlotZoomer, public TimePrecisionClass
...
+    : DisplayPlot(nplots, parent), d_xdata(1024)
 {
-    d_numPoints = 1024;
-    d_xdata = new double[d_numPoints];
-    memset(d_xdata, 0x0, d_numPoints * sizeof(double));
-
-    d_tag_text_color = Qt::black;
-    d_tag_background_color = Qt::white;
-    d_tag_background_style = Qt::NoBrush;
+    d_numPoints = d_xdata.size();
```
[the remark this answers is not in the document]
Comment (ThomasHabets) [reply gr-qtgui/lib/TimeDomainDisplayPlot.cc:99]: `d_numPoints` is not part of this class, but of the base class. Therefore can't be set by the constructor's initializer list.

**Pull request merged**